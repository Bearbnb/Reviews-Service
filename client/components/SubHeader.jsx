import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import ReviewSummaryLine from './ReviewSummaryLine.jsx';
import styles from '../../styles/ReviewSummary.css';

const SubHeader = ({searched, categories, ratingToStars, reviews, term, resetHomes}) => {
  if (searched === false) {
    return (
      <div className={styles.breakdownComponent}>
        {categories.map(category => <ReviewSummaryLine category={category} ratingToStars={ratingToStars} />)}
      </div>);
  }
  return (
    <div>
      <h4>{reviews.length} guests have mentioned " {term} "</h4>
      <button type="button" id={styles.resetReviews} onClick={() => resetHomes()}> Back to all reviews </button>
    </div>
  );
};

SubHeader.propTypes = {
  searched: PropTypes.bool.isRequired,
  categories: PropTypes.array.isRequired,
  ratingToStars: PropTypes.func.isRequired,
  reviews: PropTypes.array.isRequired,
  term: PropTypes.string.isRequired,
  resetHomes: PropTypes.func.isRequired,
};


export default SubHeader;
