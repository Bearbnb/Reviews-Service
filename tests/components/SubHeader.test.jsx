import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import SubHeader from '../../client/components/SubHeader';

describe('<SubHeader />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<SubHeader
      searched
      categories={[{}]}
      ratingToStars={jest.fn()}
      reviews={[]}
      term=""
      resetHomes={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render properly when no search', () => {
    const wrapper = shallow(<SubHeader
      searched={false}
      categories={[{}]}
      ratingToStars={jest.fn()}
      reviews={[]}
      term=""
      resetHomes={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
