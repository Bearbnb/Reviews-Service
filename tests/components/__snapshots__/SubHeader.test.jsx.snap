// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubHeader /> should render properly 1`] = `
<div>
  <h4>
    0
     guests have mentioned  "
    "
  </h4>
  <button
    id="resetReviews"
    onClick={[Function]}
    type="button"
  >
     Back to all reviews 
  </button>
</div>
`;

exports[`<SubHeader /> should render properly when no search 1`] = `
<div
  className="breakdownComponent"
>
  <ReviewSummaryLine
    category={Object {}}
    ratingToStars={[MockFunction]}
  />
</div>
`;
